src/Core/Grid/Data/Factory/CustomerBoughtProductGridDataFactoryDecorator.php

```diff
@@ -0,0 +1,102 @@
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Grid\Data\Factory;
+
+use PrestaShop\PrestaShop\Core\Grid\Data\GridData;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollection;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollectionInterface;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+use PrestaShopBundle\Service\Routing\Router;
+
+/**
+ * Class CustomerBoughtProductGridDataFactoryDecorator decorates data from customer bought products doctrine data factory.
+ */
+final class CustomerBoughtProductGridDataFactoryDecorator implements GridDataFactoryInterface
+{
+    /**
+     * @var GridDataFactoryInterface
+     */
+    private $customerBoughtProductDoctrineGridDataFactory;
+
+    /**
+     * @var Router
+     */
+    private $router;
+
+    /**
+     * @param GridDataFactoryInterface $customerBoughtProductDoctrineGridDataFactory
+     * @param Router $router
+     */
+    public function __construct(
+        GridDataFactoryInterface $customerBoughtProductDoctrineGridDataFactory,
+        Router $router
+    ) {
+        $this->customerBoughtProductDoctrineGridDataFactory = $customerBoughtProductDoctrineGridDataFactory;
+        $this->router = $router;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData(SearchCriteriaInterface $searchCriteria)
+    {
+        $customerData = $this->customerBoughtProductDoctrineGridDataFactory->getData($searchCriteria);
+
+        $records = $this->applyModifications($customerData->getRecords());
+
+        return new GridData(
+            $records,
+            $customerData->getRecordsTotal(),
+            $customerData->getQuery()
+        );
+    }
+
+    /**
+     * @param RecordCollectionInterface $records
+     *
+     * @return RecordCollection
+     */
+    private function applyModifications(RecordCollectionInterface $records)
+    {
+        $modifiedRecord = [];
+
+        foreach ($records as $r) {
+            // Set links to order detail in backoffice
+            $orderUrl = $this->router->generate('admin_orders_view', [
+                'orderId' => (int) $r['id_order'],
+            ]);
+            $r['product_name'] = '<a href="' . $orderUrl . '">' . $r['product_name'] . '</a>';
```

Comment:
Maybe this could look better with sprintf? 😉

Reply:
Resolved by using `LinkColumn`.